src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -259,18 +265,19 @@ private void WriteClientCtors(CodeWriter writer, LowLevelRestClient client, Buil
 
             bool hasKeyAuth = HasKeyAuth (context);
             bool hasTokenAuth = HasTokenAuth (context);
-            if (!hasKeyAuth && !hasTokenAuth)
-            {
-                throw new InvalidOperationException ("Has neither Key or Token credential-types?");
-            }
+
 
             if (hasKeyAuth)
             {
-                WriteConstructor(writer, client, true, context);
+                WriteConstructor(writer, client, CredentialKind.Key, context);
             }
             if (hasTokenAuth)
             {
-                WriteConstructor(writer, client, false, context);
+                WriteConstructor(writer, client, CredentialKind.Token, context);
+            }
+            if (!hasKeyAuth && !hasTokenAuth)
```

Comment:
Is it valid to have both `hasKeyAuth` and `hasTokenAuth`?

Reply:
Oh never mind - I see that it would create a ctor for each rather than one with both.

Reply:
Correct. You can have 0, 1, or both.